File: perl/001/main.pl
# ProjectEuler 000
# Perl / main.pl
# copyright (c) 2015 Susisu
#
# Perl 5.16.2
# `perl main.pl`

sub main {
    my $sum = 0;
    for (my $n = 1; $n < 1000; $n++) {
        if ($n % 3 == 0 || $n % 5 == 0) {
            $sum += $n;
        }
    }
    print "$sum\n";
}

&main;

File: ocaml/001/main.ml
(*
    ProjectEuler 001
    OCaml / main.ml
    copyright (c) 2015 Susisu

    OCaml 4.02.1
    `ocamlc main.ml -o bin/main`
*)

let rec calc n max sum =
    if n <= max then
        if n mod 3 = 0 || n mod 5 = 0 then
            calc (n + 1) max (sum + n)
        else
            calc (n + 1) max sum
    else
        sum;;

let main () =
    begin
        print_int (calc 1 999 0);
        print_newline ()
    end;;

main ();;
